From 0b5ae1fb7a8034a3560ae7b5df914092dc12bb7d Mon Sep 17 00:00:00 2001
From: Paul Eggert <eggert@cs.ucla.edu>
Date: Wed, 2 Dec 2020 15:39:41 -0800
Subject: [PATCH] Fix some errors in recent Bahamas changes

(Thanks to P Chan.)
* northamerica (Bahamas): Bahamas started DST in 1942, not 1941.
"Midnight" meant 24:00 (not 00:00) for three WWII transitions.
---
 northamerica | 13 +++++++------
 1 file changed, 7 insertions(+), 6 deletions(-)

diff --git a/northamerica b/northamerica
index 6916880..d8bb8f8 100644
--- a/northamerica
+++ b/northamerica
@@ -2935,9 +2935,10 @@ Zone America/Tijuana	-7:48:04 -	LMT	1922 Jan  1  0:11:56
 #
 # For 1899 Milne gives -5:09:29.5; round that.
 #
-# From P Chan (2020-11-27)
-# There were two periods of DST observed in 1941-1945: 1941-05-01 midnight to
-# 1944-12-31 and 1945-02-01 to 1945-10-17 midnight.
+# From P Chan (2020-11-27, corrected on 2020-12-02):
+# There were two periods of DST observed in 1942-1945: 1942-05-01
+# midnight to 1944-12-31 midnight and 1945-02-01 to 1945-10-17 midnight.
+# "midnight" should mean 24:00 from the context.
 #
 # War Time Order 1942 [1942-05-01] and War Time (No. 2) Order 1942  [1942-09-29]
 # Appendix to the Statutes of 7 George VI. and the Year 1942. p 34, 43
@@ -2961,11 +2962,11 @@ Zone America/Tijuana	-7:48:04 -	LMT	1922 Jan  1  0:11:56
 # http://www.jonesbahamas.com/?c=45&a=10412
 
 # Rule	NAME	FROM	TO	-	IN	ON	AT	SAVE	LETTER/S
-Rule	Bahamas	1941	only	-	May	 1	0:00	1:00	W
-Rule	Bahamas	1944	only	-	Dec	31	0:00	0	S
+Rule	Bahamas	1942	only	-	May	 1	24:00	1:00	W
+Rule	Bahamas	1944	only	-	Dec	31	24:00	0	S
 Rule	Bahamas	1945	only	-	Feb	 1	0:00	1:00	W
 Rule	Bahamas	1945	only	-	Aug	14	23:00u	1:00	P # Peace
-Rule	Bahamas	1945	only	-	Oct	17	0:00	0	S
+Rule	Bahamas	1945	only	-	Oct	17	24:00	0	S
 Rule	Bahamas	1964	1975	-	Oct	lastSun	2:00	0	S
 Rule	Bahamas	1964	1975	-	Apr	lastSun	2:00	1:00	D
 # Zone	NAME		STDOFF	RULES	FORMAT	[UNTIL]
-- 
1.8.3.1

